app/code/community/TIG/MyParcel2014/Model/Api/MyParcel.php

```diff
@@ -803,35 +791,23 @@ protected function _getConsignmentData(TIG_MyParcel2014_Model_Shipment $myParcel
                     }
                 }
             }
-            $data['customs_declaration']['weight'] = (int)$totalWeight;
-            $data['physical_properties']['weight'] = (int)$totalWeight;
+            $data['customs_declaration']['weight'] = (int)$WeightData[1];
+
         }
 
         if($data['options']['package_type'] == self::DIGITAL_STAMP){
-            $totalWeight = 0;
-            $items = $myParcelShipment->getOrder()->getAllItems();
             foreach($items as $item) {
                 if($item->getProductType() == 'simple') {
-                    $parentId = $item->getParentItemId();
-                    $weight   = floatval($item->getWeight());
-                    $qty      = intval($item->getQtyOrdered());
-
-                    if ( ! empty($parentId)) {
-                        $parent = Mage::getModel('sales/order_item')->load($parentId);
-
-                        if (empty($weight)) {
-                            $weight = $parent->getWeight();
-                        }
-                    }
-                    $weight      *= $qty;
-                    $totalWeight += $weight * 1000;
+                    $WeightData = $this->getTotalWeight($totalWeight, $item);
```

Comment:
[$weight, $totalWeight, $qty, $price] = $this->getTotalWeight($totalWeight, $item);

Reply:
Done!